package/network/config/gre/files/gre.sh

```diff
@@ -101,16 +114,29 @@ proto_gretap_setup() {
 grev6_setup() {
 	local cfg="$1"
 	local mode="$2"
+	local remoteip6
 
 	local ip6addr peer6addr weakif
 	json_get_vars ip6addr peer6addr tunlink weakif
 
 	[ -z "$peer6addr" ] && {
-		proto_notify_error "$cfg" "MISSING_ADDRESS"
+		proto_notify_error "$cfg" "MISSING_PEER_ADDRESS"
 		proto_block_restart "$cfg"
 		exit
 	}
 
+	remoteip6=$(resolveip -t 10 -6 $peer6addr)
```

Comment:
Please quote the `$peer6addr` here to avoid shell injections with invalid input.